Convert to

valeriereyes5 opened the pull request.

valeriereyes5 committed:
Created feature class, and implemented Converto

Implemented converTo and helper methods. Still needs improvement.

valeriereyes5 committed:
Update feature.py

Started fixing convertTo of weight related units

valeriereyes5 committed:
Update feature.py

Fixed weigth related convert to

valeriereyes5 committed:
Update feature.py

Fixed converTo parameters

isaacrivriv commented on src/feature.py:
```diff
@@ -0,0 +1,134 @@
+import sys
+# insert at 1, 0 is the script path (or '' in REPL)
+sys.path.insert(1, '/Users/valericita/Documents/GitHub/ChemLAB-PL/src/chem_lex')
```
We cant have these types of paths in here. This path is only available at your computer so it wont work on any computer. We need to change this to another way

isaacrivriv commented:
If it works Im okay with it. Theres two issues we need to address before merging.
- Make the change I mentioned about the path
- Add the functionality to the parser

valeriereyes5 committed:
Fixed Import problem

Added an empty __init__ file to the parser folder to fix it

isaacrivriv committed:
Merge branch 'master' into ConvertTo

valeriereyes5 committed:
Fixed feature and implemented it on  parser

fixed feature and tried implementing it into thr parser

valeriereyes5 committed:
Implemented in parser and feature

Implemented the convertTo in parser, fixed the prefix convert To in feature.

isaacrivriv commented on src/feature.py:
```diff
@@ -0,0 +1,169 @@
+def convertTo(el, num, Unit, newUnit):
+    ### need to redo with the parser
+
+    if(Unit in unit_prefix and Unit in unit_prefix):
```
It's comparing the same thing twice? Think something may be wrong here. Clean up redundancy

isaacrivriv commented on src/chem_parse/ChemlabParser.py:
```diff
@@ -146,7 +147,8 @@ def p_built_in(self, p):
                         p[8]) + " ; to: " + str(p[10]))
-            p[0]["value"] = 24.6
+
+            p[0]["value"] = feature.convertTo(p[3],p[6],(p[8])[0],(p[10])[0])
```
We have two rules that apply to conversion in the grammar. We are not taking it into account. We need to fix this

isaacrivriv replied:
Also where do you pass the prefix in here? Is it mixed with the unit?

isaacrivriv commented on src/chem_parse/ChemlabParser.py:
```diff
@@ -229,7 +231,8 @@ def p_form_manage(self, p):
     def p_unit(self, p):
         '''Unit : UnitTok
-                        | PrefixTok Unit'''
+                        | PrefixTok
```
Im okay with just passing the prefix but we really need to test this well.

isaacrivriv replied:
Also you changed the rule but did not update the logic. I think you meant to only pass in either a unit, a prefix, or a prefix followed by a unit, not the unit rule but the unit token

isaacrivriv commented on src/feature.py:
```diff
@@ -0,0 +1,169 @@
+from chem_lex.ChemlabTokens import unit_prefix, supported_units
+#import
+from element import Element
+def convertTo(el, num, Unit, newUnit):
```
I need you to test this with the chemlab test included and make sure that all still passes and that the tests in there dont blow up

isaacrivriv commented on src/feature.py:
```diff
@@ -0,0 +1,169 @@
+
+        result = convertPrefix(num,Unit,newUnit)
+        print("Result:    " + str(result)+" "+ newUnit)
+    elif (Unit in supported_units and Unit in supported_units):
```
Again redundancy found

isaacrivriv commented on src/feature.py:
```diff
@@ -0,0 +1,169 @@
+    return result
+
+
+pre = {'G': 10**9,  # GIGA
```
Im not sure how the prefixes are being used cause Im not sure how they are passed in. We need to talk about this a bit

isaacrivriv reviewed (changes requested):
So there a couple of things I found that seem pretty weird. Lets talk a bit about this

valeriereyes5 committed:
Update feature.py

Fixed redundancy in if statements

valeriereyes5 committed:
Update ChemlabParser.py

changed Unit definition

isaacrivriv commented on src/feature.py:
```diff
@@ -0,0 +1,169 @@
+}
+def convertPrefix(num,Unit,newUnit):
+    CNum = num * pre[Unit]
+    if (newUnit =='G'):
```
Change all the ifs either to elif or even better use the pre dictionary and divide with it

isaacrivriv reviewed (commented):
The other comment we mentioned was with the changes to the lexer to not combine the prefixes and units

valeriereyes5 committed:
Fixed feature,parser and tokens/lexer

Implemented prefix and unit in convertTo, Removed prefix and unit combination, Fixed unit definition in parser

isaacrivriv committed:
Small changes regarding parsing rules

isaacrivriv reviewed (approved):
Looks good. Made small changes when testing. Merging
